tests/test_serialization.py

```diff
@@ -217,3 +224,93 @@ def test_serialize_fail_llla_different_last_layer_name(laplace, model, model3, r
 
     with pytest.raises(ValueError):
         la2.load_state_dict(torch.load('state_dict.bin'))
+
+
+@pytest.mark.parametrize(
+    "model_device_str",
+    ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"],
+)
+@pytest.mark.parametrize(
+    "map_location",
+    ["cuda", "cpu"] if torch.cuda.is_available() else ["cpu"],
+)
+@pytest.mark.parametrize("laplace", flavors + flavors_subnet)
+def test_map_location(
+    laplace, model, reg_loader, tmp_path, map_location, model_device_str
+):
+    # Only for LLLaplace, we need dill.
+    #   AttributeError: Can't pickle local object 'FeatureExtractor._get_hook.<locals>.hook'
+    if issubclass(laplace, LLLaplace):
+        if find_spec("dill") is None:
+            pytest.skip(
+                reason="dill package not found but needed for this test"
+            )
+        else:
+            import dill
+
+            torch_save = lambda obj, fn: torch.save(
+                obj, fn, pickle_module=dill
+            )
+    else:
+        # Use default pickle_module=pickle, but no need to import pickle here
+        # just to set pickle_module=pickle.
+        torch_save = lambda obj, fn: torch.save(obj, fn)
+
+    device = torch.device(model_device_str)
+    kwds = dict(model=model.to(device), likelihood="regression")
+    if issubclass(laplace, SubnetLaplace):
+        kwds.update(subnetwork_indices=torch.LongTensor([1, 10, 104, 44]))
+    la = laplace(**kwds)
+
+    # device(type='cuda', index=0) != device(type='cuda')
+    ##assert la._device == device
+    assert la._device.type == device.type
+
+    la.fit(reg_loader)
+    la.optimize_prior_precision()
+    la.sigma_noise = 1231
+
+    save_fn = tmp_path / "la.pt"
```

Comment:
Does this line create an actual file? If so, it should be cleaned up after this unit test finishes. You can add this file to the `cleanup` function above in this file.

Reply:
It uses the `pytest` [`tmp_path` fixture])(https://docs.pytest.org/en/latest/reference/reference.html#tmp-path) so the file is subject to `pytest`'s file retention policy.

Reply:
Ok good to know!